pageserver/src/tenant/layer_map.rs

```diff
@@ -53,15 +53,27 @@ pub struct LayerMap {
     pub frozen_layers: VecDeque<Arc<InMemoryLayer>>,
 
     /// All the historic layers are kept here
-    historic_layers: RTree<LayerRTreeObject>,
+    historic_layers: RTree<LayerRTreeObject<PersistentLayerT>>,
 
     /// L0 layers have key range Key::MIN..Key::MAX, and locating them using R-Tree search is very inefficient.
     /// So L0 layers are held in l0_delta_layers vector, in addition to the R-tree.
-    l0_delta_layers: Vec<Arc<dyn Layer>>,
+    l0_delta_layers: Vec<Arc<PersistentLayerT>>,
+}
+
+impl<T: ?Sized> Default for LayerMap<T> {
+    fn default() -> Self {
+        Self {
+            open_layer: Default::default(),
```

Comment:
> Default::default

IMO, it's quite unreadable and also gets the compiler to do some extra type inference work that it can avoid.
We could use actual container types instead of `Default`: `None`, `VecDeque::default/new`, etc.